File: src/javascript/game_of_life.js
class Board 
{
    constructor(size)
    {
        this.Size = size
        this.Grid = new Array(size).fill(null).map(() => new Array(size).fill(false))
    }

    Print()
    {
        for (let i = 0; i < this.Size; i++)
        {
            for (let j = 0; j < this.Size; j++)
            {
                process.stdout.write(this.Grid[i][j] ? '█' : '░')
            }
            process.stdout.write('\n')
        }
    }

    ComputeNextGeneration()
    {
        const newGrid = new Array(this.Size).fill(null).map(() => new Array(this.Size).fill(false))
        let hasChanged = false

        for (let i = 0; i < this.Size; i++)
        {
            for (let j = 0; j < this.Size; j++)
            {
                const aliveNeighbours = this.CountAliveNeighbours(i, j)
                newGrid[i][j] = this.Grid[i][j] ? (aliveNeighbours === 2 || aliveNeighbours === 3) : aliveNeighbours === 3

                if (newGrid[i][j] !== this.Grid[i][j])
                    hasChanged = true
            }
        }

        this.Grid = newGrid
        return hasChanged
    }

    CountAliveNeighbours(i, j)
    {
        let count = 0
        for (let x = -1; x <= 1; x++)
        {
            for (let y = -1; y <= 1; y++)
            {
                if (x === 0 && y === 0)
                    continue

                const neighbourI = i + x
                const neighbourJ = j + y
                if (neighbourI >= 0 && neighbourI < this.Size && neighbourJ >= 0 && neighbourJ < this.Size)
                    count += this.Grid[neighbourI][neighbourJ] ? 1 : 0
            }
        }
        return count
    }

}

const BOARD_SIZE = 20
const FPS = 15, TICKS = 1000 / FPS

const board = new Board(BOARD_SIZE)

for (let i = 0; i < BOARD_SIZE; i++)
    for (let j = 0; j < BOARD_SIZE; j++)
        board.Grid[i][j] = Math.random() < 0.5

console.clear();
main()
async function main()
{
    while (board.ComputeNextGeneration())
    {
        board.Print()
        process.stdout.cursorTo(0, 0)
        await new Promise(resolve => setTimeout(resolve, TICKS))
    }
}

File: src/python/game_of_life.py
import random
import time
import os
import sys

class Board:
    Size: int
    Grid: list

    def __init__(self, size):
        self.Size = size
        self.Grid = [[False for _ in range(size)] for _ in range(size)]
    
    def Print(self):
        for i in range(self.Size):
            for j in range(self.Size):
                if self.Grid[i][j]:
                    sys.stdout.write('█')
                else:
                    sys.stdout.write('░')
            print()

    def ComputeNextGeneration(self) -> bool:
        new_grid = [[False for _ in range(self.Size)] for _ in range(self.Size)]
        hasChanged = False

        for i in range(self.Size):
            for j in range(self.Size):
                aliveNeighbours = self.CountAliveNeighbours(i, j)
                if self.Grid[i][j]:
                    new_grid[i][j] = aliveNeighbours == 2 or aliveNeighbours == 3
                else:
                    new_grid[i][j] = aliveNeighbours == 3

                if new_grid[i][j] != self.Grid[i][j]:
                    hasChanged = True

        self.Grid = new_grid
        return hasChanged

    def CountAliveNeighbours(self, i, j) -> int:
        count = 0
        for x in range(-1, 2):
            for y in range(-1, 2):
                if x == 0 and y == 0:
                    continue
                neighbourI = i + x
                neighbourJ = j + y
                if neighbourI >= 0 and neighbourI < self.Size and neighbourJ >= 0 and neighbourJ < self.Size:
                    if self.Grid[neighbourI][neighbourJ]:
                        count += 1
        return count

BOARD_SIZE = 20
FPS = 15
TICKS = 1000/FPS / 1000 # Python sleep() uses seconds

board = Board(BOARD_SIZE)

for i in range(BOARD_SIZE):
    for j in range(BOARD_SIZE):
        board.Grid[i][j] = random.randint(0, 1) < 0.5

os.system('cls') if os.name == 'nt' else os.system('clear')
while(board.ComputeNextGeneration()):
    print('\033[0;0H')
    board.Print()
    time.sleep(TICKS)